label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRule data = 2;
+}
+
+message CreateLabelApplicationRule {
+  LabelApplicationRule data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  string description = 2;
+  MatchingCriteria matching_criteria = 3;
+  ApplicationCriteria application_criteria = 4;
+
+  message MatchingCriteria {
+    Type type = 1;
+    repeated Condition conditions = 2;
+
+    enum Type {
+      TYPE_UNSPECIFIED = 0;
+      TYPE_SPAN = 1;
+      TYPE_TRACE = 2;
+    }
+
+    message Condition {
+      string name = 1;
+      Scope scope = 2;
+      string key = 3;
+      Operator operator = 4;
+      optional string value = 5;
+
+      enum Scope {
+        SCOPE_UNSPECIFIED = 0;
+        SCOPE_ATTRIBUTE = 1;
+        SCOPE_REQUEST = 2;
+        SCOPE_RESPONSE = 3;
+      }
+
+      enum Operator {
+        OPERATOR_UNSPECIFIED = 0;
+        OPERATOR_EXISTS = 1;
+        OPERATOR_EQUALS = 2;
+        OPERATOR_CONTAINS = 3;
+        OPERATOR_STARTS_WITH = 4;
+        OPERATOR_ENDS_WITH = 5;
+        OPERATOR_MATCHES_REGEX = 6;
+      }
+    }
+  }
+
+  message ApplicationCriteria {
+    Type type = 1;
+    Scope scope = 2;
+    Action action = 3;
+
+    enum Type {
+      TYPE_UNSPECIFIED = 0;
+      TYPE_ENTITY = 1;
+    }
+
+    enum Scope {
+      SCOPE_UNSPECIFIED = 0;
+      SCOPE_API = 1;
+      SCOPE_SERVICE = 2;
+      SCOPE_DOMAIN = 3;
+    }
+
+    message Action {
+      Operation operation = 1;
+      oneof value {
+        string static_label_id = 2;
+        DynamicLabel dynamic_label = 3;
+      }
+
+      enum Operation {
+        OPERATION_UNSPECIFIED = 0;
+        OPERATION_MERGE = 1;
+      }
+
+      message DynamicLabel {
+        repeated TokenExtractionRule token_extraction_rules = 1;
+        string token_combining_expression  = 2;
```

Comment:
What's the expression syntax? Do we even need the token rules if alias is optional?

Reply:
I was thinking of format supported by slf4j MessageFormatter

Reply:
We can get rid of token extraction rule completely and just have `label_expression` which can be of format `${deployment}_${version}`. I forgot to add whether we need to extract token from request/response/attribute. If we are just going with attribute, then `TokenExtractionRule` is probably not needed.

Reply:
So the above example would apply the label "mydeployment-1.3" if the span had below attributes? Just want to make sure I understand correctly:
```json
{
  "deployment": "mydeployment",
  "version": "1.3"
}
```

Reply:
Yes